src/dswx_sar/save_mgrs_tiles.py

```diff
@@ -88,41 +127,179 @@ def save_mgrs_tile(source_tif_path,
+    x_value_min, x_value_max, y_value_min, y_value_max, epsg_output = \
+        get_geographic_boundaries_from_mgrs_tile(output_mgrs_id,
+                                                 mgrs_db_path,)
 
-    # Compute bounding extents for given MGRS tile
-    x_value_min, x_value_max, y_value_min, y_value_max = \
-        get_geographic_boundaries_from_mgrs_tile(output_mgrs_id)
     bbox = [x_value_min, y_value_min, x_value_max, y_value_max]
 
     # create MGRS tiles for products.
     output_tif_file_path = f'{output_dir_path}/{output_tif_name}'
-    ds = input_tif_obj
     opt = gdal.WarpOptions(dstSRS=f'EPSG:{epsg_output}',
-                     outputType=band.DataType,
-                     xRes=xspacing,
-                     yRes=yspacing,
-                     outputBounds=bbox,
-                     resampleAlg=method,
-                     dstNodata=no_data_value,
-                     format='GTIFF')
+                           outputType=band.DataType,
+                           xRes=xspacing,
+                           yRes=yspacing,
+                           outputBounds=bbox,
+                           resampleAlg=method,
+                           format='GTIFF')
     gdal.Warp(output_tif_file_path, source_tif_path, options=opt)
-    dswx_sar_util._save_as_cog(output_tif_file_path, output_dir_path, logger,
-                compression='ZSTD',
-                nbits=16)
+    input_tif_obj = None
+    dswx_sar_util._save_as_cog(output_tif_file_path,
+                               output_dir_path,
+                               logger,
+                               compression='ZSTD',
+                               nbits=16)
+
+
+def find_mgrs_collection(image_tif,
+                         mgrs_collection_file=None):
+    """Find MGRS tile collection intersecting Geotiff
+    Parameters
+    ----------
+    image_tif: str
+        path of input geotiff
+    mgrs_collection_file : str
+        path for MGRS tile collection
+
+    Returns
+    ----------
+    mgrs_list: list
+        list of MGRS tiles
+    """
+    if mgrs_collection_file is not None:
+        # Load the raster data
+        with rasterio.open(image_tif) as src:
+            epsg_code = int(src.crs.data['init'].split(':')[1])
+
+            # Get bounds of the raster data
+            left, bottom, right, top = src.bounds
+
+            # Reproject to EPSG 4326 if the current EPSG is not 4326
+            if epsg_code != 4326:
+                left, bottom, right, top = transform_bounds(
+                                                    src.crs,
+                                                    {'init': 'EPSG:4326'},
+                                                    left,
+                                                    bottom,
+                                                    right,
+                                                    top)
+
+        # Create a Polygon from the bounds
+        raster_polygon = Polygon([(left, bottom),
```

Comment:
If there's antimeridian crossing, the `right` coordinate may be smaller than the `left` coordinate, e.g., -179 vs +179, respectively. If that happens, you may need to add `360` to the `right` coordinate. So that your polygon is not "huge".